inspirehep/dojson/hepnames/fields/bd1xx.py

```diff
@@ -126,9 +131,29 @@ def name2marc(self, key, value):
 @utils.for_each_value
 def ids(self, key, value):
     """All identifiers, both internal and external."""
+    IDS_MAP = {
+        'ARXIV': 'ARXIV',
+        'BAI': 'INSPIRE BAI',
+        'CERN': 'CERN',
+        'DESY': 'DESY',
+        'INSPIRE': 'INSPIRE ID',
+        'KAKEN': 'KAKEN',
+        'ORCID': 'ORCID',
+        'SLAC': 'SLAC',
+        'WIKIPEDIA': 'WIKIPEDIA',
+    }
+
+    type_ = IDS_MAP.get(value.get('9').upper())
+    a_value = value.get('a')
+
+    if type_ == 'CERN' and a_value.startswith('CERN-'):
+        a_value = 'CERN-' + NON_DIGIT.sub('', a_value)
```

Comment:
Mmh. Here you would happily normalize crap if `a_value` contained crap. You could simply skip this identifier if it doesn't match `CERN-\d+`, and otherwise do `a_value[len('CERN-'):]` to strip the right length.